### DISCUSSION TITLE
Rotating secrets that were formerly used to bootstrap flux

### DISCUSSION THREAD

**Discussion by kuratowskiy** [fluxcd/flux2#1159] [Q&A] [answered]
My understanding is that all of the secrets that I reference in my Kustomizations can be rotated easily in the GitOps way, with the exception of the one that is used to bootstrap the cluster.

How can this particular secret be managed by Flux, so that it can be possible to just commit a new one somewhere whenever it is time for rotation?

**Comment by hiddeco** [comment, marked as answer]
You can simply commit one to Git, the Kustomization will then overwrite it on the next reconciliation and use it for reconciliations that follow. Once the secret is present in the cluster, `flux bootstrap` will not try to generate a new one but simply assume that the credentials in the cluster work.

Once #968 is merged, it will even be possible to supply your own private key to the bootstrap using the `--private-key-file` flag, although the bootstrap process will not commit it to Git but only apply it to the cluster (and configure it for the repository).

#968 will also make it possible to have control over the name of the secret using `--secret-name`, instead of having the hardcoded default that equals to the namespace Flux is bootstrapped to (`flux-system` without supplying a `--namespace` param).

**Reply by kuratowskiy** [reply]
Thanks
